src/PureCloud.Client/Apis/VoicemailApi.cs

```diff
@@ -0,0 +1,97 @@
+using System.Net.Http.Json;
+using Microsoft.Extensions.Options;
+using PureCloud.Client.Contracts;
+using PureCloud.Client.Http;
+using PureCloud.Client.Json;
+using PureCloud.Client.Models;
+
+namespace PureCloud.Client.Apis;
+
+public sealed class VoicemailApi : IVoicemailApi
+{
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly PureCloudJsonSerializerOptions _options;
+
+    public VoicemailApi(
+        IHttpClientFactory httpClientFactory,
+        IOptions<PureCloudJsonSerializerOptions> options)
+    {
+        _httpClientFactory = httpClientFactory;
+        _options = options.Value;
+    }
+
+    /// <inheritdoc />
+    public async Task<VoicemailMessage> GetVoicemailMessageAsync(string messageId, IEnumerable<string> expand = null, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(messageId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = $"api/v2/voicemail/messages/{messageId}";
+        if (expand != null)
+        {
+            var expandParam = UriHelper.ParameterToString(expand);
+            uri += $"?expand={expandParam}";
+        }
```

Comment:
build Uri using UriHelper.GetUrl method. Also escape each parameter line messageId before using it.